PlaceRouteHierFlow/router/A_star.cpp

```diff
@@ -0,0 +1,2515 @@
+                      current_node = grid.vertices_total[current_node].up;
+                      h_lock = 1;
+                      if(current_node==-1){return 0;}
+                     }else{
+                      current_node = grid.vertices_total[current_node].down;
+                      h_lock = 1;
+                      if(current_node==-1){return 0;}
+                     }
+*/
+                 }else if(v_search_temp == v_search){
+                   
+                    if(v_search_temp=1){
+
+                       int current_x = grid.vertices_total[current_node].x;
+                       current_node = current_node + 1;
+                       if(current_node>grid.vertices_total.size()){return 0;}
+                       int next_x = grid.vertices_total[current_node].x;
+                       if(current_x!=next_x){return 0;}
+
+                      }else{
+
+                       int current_x = grid.vertices_total[current_node].x;
+                       current_node = current_node -1;
+                       if(current_node<0){return 0;}
+                       int next_x = grid.vertices_total[current_node].x;
+                       if(current_x!=next_x){return 0;}
+
+                      }
+
+                 }else{
+ 
+                       return 0;
+               
+                 }
+
+              }
+          }
+
+    }
+
+  return 1;
+
+
+};
+
+//you xuan connection rule
+bool A_star::Check_Connection_Rule(std::vector<int> &left_up1, std::vector<int> &right_up1, std::vector<int> &left_up2, std::vector<int> &right_up2, Grid& grid){
+
+   for(int i=0;i<left_up1.size();i++){
+
+       if(!L_shape_Connection(left_up1[i],left_up2[i],grid)){return 0;}
+
+      }
+
+   for(int i=0;i<right_up1.size();i++){
+
+       if(!L_shape_Connection(right_up1[i],right_up2[i],grid)){return 0;}
+
+      }
+
+   return 1;
+
+};
+
+std::vector<int> A_star::A_star_algorithm(Grid& grid, int left_up, int right_down){
+
+  // how to back trace?
+
+  int via_expand_effort = 100;
+
+  //std::cout<<"A start checkout point1"<<std::endl;
+
+  std::set<std::pair<int,int>, RouterDB::pairComp> L_list;
+  std::pair<int,int> temp_pair; //temp_pair.first is the cost, temp_pair.second is the index of graph
+
+  std::set<int> src_index;
+  for(int i=0;i<source.size();i++){
+    
+      src_index.insert(source[i]);
+
+     }
+  
+  std::set<int> dest_index;
+  for(int i=0;i<dest.size();i++){
+    
+      dest_index.insert(dest[i]);
+
+     }
+
+  //initial source  Q1//jugde whether source works
+  //initial_source(grid, L_list);
+  initial_source(grid, L_list, src_index, left_up, right_down);
+
+  bool found = 0;
+  int current_node = -1;
+
+  //std::cout<<"A start checkout point2"<<std::endl;
+  
+  while(!L_list.empty() and !found){
+    //std::cout<<"L_list size"<<L_list.size()<<std::endl;
+    std::set<std::pair<int,int>, RouterDB::pairComp>::iterator it;
+    it = L_list.begin();
+    L_list.erase(it);
```

Comment:
Codacy is complaining about erasing the value at the iterator and then using the iterator in the next line.
You might want to double check that what you are doing is okay.

Reply:
@stevenmburns 

Thanks.